Avoid PHP 8.2+ deprecation warnings on creating dynamic properties

esszett opened the pull request:
Adding explicit declarations for the properties:
public $uninstall 
public $submit_save_module
Both used by ProcessWire's module system during configuration saving

esszett committed:
Avoid PHP 8.2+ deprecation warnings on creating dynamic properties

Adding explicit declarations for the properties:
public $uninstall 
public $submit_save_module
Both used by ProcessWire's module system during configuration saving

MoritzLost commented:
@esszett Thanks for the PR!

Happy to merge this, but I think the underlying problem is that the module class extends `Wire` instead of `WireData`. According to the [module docs](https://processwire.com/api/ref/module/), modules should usually extend the `WireData` class. I think that would get rid of the deprecation warnings, because that class has the magic getter/setter methods required for dynamic properties. Might be easier to just change the module to extend `WireData` instead of manually adding all the dynamic properties that ProcessWire may want to add.

What do you think? Can you give this a try?

esszett commented:
Ah that's a good point – I overlooked that. I'll give it a try and report back.

esszett commented:
Hello Moritz, I can confirm that everything is working as expected with the ‘extends WireData’ adjustment (PHP 8.2). I was unable to reproduce my original deprecation warning. I did edit my pull request accordingly.

esszett committed:
Update CachePlaceholders.module

Revert adding property declarations. Instead change class 'extend Wire' to recommended 'extend WireData'.

MoritzLost commented:
@esszett Thanks for testing and confirming that! I've merged the PR, updated the changelog and bumped the version number. Thanks for the contribution!
